### PR TITLE
Avoid duplication of `--set-host-id-source` OneAgent argument

### PR DESCRIPTION
**Pull request by StefanHauth**
## Description

If the OneAgent argument `--set-host-id-source` is provided in the DynaKube, it causes a duplication for classic fullstack and host monitoring modes. It's then unclear which argument counts. 

This PR fixes the handling of this argument, so that it can be overridden in the DynaKube while still maintaining the applied default values otherwise.

## How can this be tested?

* deploy an ClassicFull stack Dynakube with and without `--set-host-id-source=fqdn` in it
* check the OneAgent pod specs, they should contain it only once (with value `fqdn` resp. `auto` as default)

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: pkg/controllers/dynakube/oneagent/daemonset/arguments.go [hunk 1]
```diff
@@ -57,3 +64,7 @@ func (dsInfo *builderInfo) appendProxyArg(argMap *prioritymap.Map) {
 func (dsInfo *builderInfo) hasProxy() bool {
 	return dsInfo.dynakube != nil && dsInfo.dynakube.NeedsOneAgentProxy()
 }
+
+func appendHostIdSource(argMap *prioritymap.Map, hostIdSource string) {
```
Comment (andriisoldatenko) [inline pkg/controllers/dynakube/oneagent/daemonset/arguments.go:68]: I don't see any benefits of using function for 1 line to replace one line .

**Comment by andriisoldatenko** [comment]
practically LGTM.

**Review by aorcholski — CHANGES_REQUESTED** [review]

File: pkg/controllers/dynakube/oneagent/daemonset/arguments_test.go [hunk 2]
```diff
@@ -58,6 +58,29 @@ func TestArguments(t *testing.T) {
 		assert.NotEmpty(t, podSpecs.Containers)
 		assert.Contains(t, podSpecs.Containers[0].Args, testValue)
 	})
+	t.Run("classic fullstack", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (StefanHauth) [reply pkg/controllers/dynakube/oneagent/daemonset/arguments_test.go]: fixed in [37f5af7](https://github.com/Dynatrace/dynatrace-operator/pull/2508/commits/37f5af762bc2c0bdef231ee7781663376055dbed)

File: pkg/controllers/dynakube/oneagent/daemonset/arguments.go (hunk 1 above)
Comment (StefanHauth) [reply pkg/controllers/dynakube/oneagent/daemonset/arguments.go]: fixed in [37f5af7](https://github.com/Dynatrace/dynatrace-operator/pull/2508/commits/37f5af762bc2c0bdef231ee7781663376055dbed)

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**